src/meteodatalab/grib_decoder.py

```diff
@@ -379,13 +398,21 @@ def save(
         if (dim := str(key)) not in {"x", "y"}
     }
 
+    step_unit = UnitOfTime.MINUTE
+    time_range_unit = UnitOfTime(md.get("indicatorOfUnitForTimeRange", 255)).unit
+    time_range = _to_timedelta(md.get("lengthOfTimeRange", 0), unit=time_range_unit)
+
     def to_grib(loc: dict[str, xr.DataArray]):
         grib_loc = {
             DIM_MAP[key]: value.item()
             for key, value in loc.items()
-            if key != "ref_time"
+            if key not in {"ref_time", "lead_time"}
         }
+        step_end = np.timedelta64(loc["lead_time"].item(), "ns")
+        step_begin = step_end - time_range
         return grib_loc | {
+            "indicatorOfUnitOfTimeRange": step_unit.value,
+            "forecastTime": step_begin / _to_timedelta(1, step_unit.unit),
```

Comment:
Could you elaborate why forecast time (previous lead time) is (leadtime - time range)? So I am interested in forecast after one timestep (step=1, leadtime=1) the forecasttime will be the same as the ref_time...?

Reply:
The `forecastTime` (GRIB key) is defining the start of the time range interval for statistical process. The `lead_time` coordinate in our case defines the end of that interval. The `forecastTime` is not necessarily the same as the previous lead time. It is only the same when the length of the time interval matches the lead time increment between time steps. The `forecastTime` may match with the reference time if that is the start of the interval.

Reply:
Ok, I understand it is not the same when it's for example an aggregation since the beginning, weighted difference for a given delta in time,...